### PR TITLE
Identity server integration

### PR DESCRIPTION
**Pull request by rudreshgp**
Added Bearer authentication using IdentityServer4. 
Added a sample project with sample users and clients to run the application.
Added documentation for the same.

### CODE REVIEW

File: src/DeviceManager.Api/appsettings.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "DatabaseType": "MsSql",
 
     // Sql Database
-    "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=DeviceDb;Trusted_Connection=True;MultipleActiveResultSets=true"
+    "DefaultConnection": "Server=192.168.96.1,1433;Database=DeviceDb;User Id=sa;Password=Abc@1234;Trusted_Connection=True;MultipleActiveResultSets=true"
```
Comment (rudreshgp) [inline src/DeviceManager.Api/appsettings.json]: Please don't merge this line. I mistakenly checked in this line.

Comment (Boriszn) [reply src/DeviceManager.Api/appsettings.json]: Hey @rudreshgp Did you mean not use this one: `"DefaultConnection": "Server=192.168.96.1,1433;Database=DeviceDb;User Id=sa;Password=Abc@1234;Trusted_Connection=True;MultipleActiveResultSets=true"`

Comment (rudreshgp) [reply src/DeviceManager.Api/appsettings.json]: Yes. 
Also, I was on holiday from the last month. I will try to contribute from now on.

Comment (Boriszn) [reply src/DeviceManager.Api/appsettings.json]: Hey @rudreshgp Ah ok. Nice to hear that.

**Pull request merged**